dislib/classification/rf/data.py

```diff
@@ -0,0 +1,199 @@
+from numpy.lib import format
+import numpy as np
+import tempfile
+
+from pycompss.api.parameter import FILE_IN, FILE_INOUT
+from pycompss.api.task import task
+
+from dislib.data import Dataset
+
+
+@task(labels_path=FILE_IN, returns=3)
+def get_labels(labels_path):
+    y = np.genfromtxt(labels_path, dtype=None, encoding='utf-8')
+    def_intp = np.intp  # Save default np.intp
+    np.intp = np.int8  # Modify np.intp, so that codes.dtype=int8
+    categories, codes = np.unique(y, return_inverse=True)
+    np.intp = def_intp  # Reset np.intp to default value
+    return codes, categories, len(categories)
+
+
+class NpyFile(object):
+    def __init__(self, path):
+        self.path = path
+
+        self.shape = None
+        self.fortran_order = None
+        self.dtype = None
+
+    def get_shape(self):
+        if self.shape is None:
+            self._read_header()
+        return self.shape
+
+    def get_fortran_order(self):
+        if self.fortran_order is None:
+            self._read_header()
+        return self.fortran_order
+
+    def get_dtype(self):
+        if self.dtype is None:
+            self._read_header()
+        return self.dtype
+
+    def _read_header(self):
+        with open(self.path, 'rb') as fp:
+            version = format.read_magic(fp)
+            try:
+                format._check_version(version)
+            except ValueError:
+                raise ValueError('Invalid file format.')
+            header_data = format._read_array_header(fp, version)
+            self.shape, self.fortran_order, self.dtype = header_data
+
+
+class RfDataset(object):
+    """
+    A dataset formatted for the random forest algorithm of this library.
+    """
+    def __init__(self, samples_path, labels_path, features_path=None):
+        """
+        Constructor for RfDataset.
+
+        :param samples_path:  Path of the .npy file containing the 2-d array of
+                              samples. Must be a str or a future COMPSs object.
+                              If it is the later, self.n_samples and
+                              self.n_features must be set manually (they can be
+                              future objects).
+        :param labels_path:   Path of the .dat file containing the 1-d array of
+                              labels. Must be a str or a future COMPSs object.
+        :param features_path: Optional. Path of the .npy file containing the
+                              2-d array of samples transposed. The array must
+                              be C-ordered. Providing this array may improve
+                              the performance as it allows sequential access to
+                              the features.
+        """
```

Comment:
We are following Numpy's docstrings format: [https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html](url)

Also, better say PyCOMPSs instead of COMPSs. Or even better, give the module name: pycompss....FutureObject

Reply:
Done.

Reply:
Can you change the class doc to something like this? https://github.com/scikit-learn/scikit-learn/blob/55bf5d9/sklearn/svm/classes.py#L429

Note that the constructor does not have docstrings, and that classes have methods and attributes.

Reply:
Done.